docs/rfcs/0561-list-with-metadata.md

```diff
@@ -0,0 +1,72 @@
+- Proposal Name: `list_with_metadata`
```

Comment:
Please use the same template for RFC.

Reply:
FYI: https://github.com/datafuselabs/opendal/blob/main/docs/rfcs/0000-example.md